packages/swayze/lib/src/core/viewport_context/viewport_context.dart

```diff
@@ -408,3 +414,49 @@ extension OverflowViewportMethods on OffscreenDetails {
   /// Defines if a [OffscreenDetails] describes a overflow situation.
   bool get isOffscreen => this != OffscreenDetails.noOverflow;
 }
+
+/// The result of the evaluation of a position on the table.
+///
+/// See also:
+/// - [ViewportContext.evaluateHover] that generates this result.
+@immutable
+class EvaluateHoverResult {
```

Comment:
Looking at the doc on the `evaluateHover` it seems like we want this method to be able to provide info for future features. So, having that in mind, this class could probably be abstract and move the `fillRange` to a different class that extends this one.

 it seems like we want to this a abstract class with a implementation specific for fill range. IE,

Reply:
Not sure if I got this. We have the viewportContext that is an abstract class, that receives an offset and should return the result with some pre-determined data. And then we have the implementation on the viewport context provider, that does the default data implementation. If someone wants, they can implement the viewport context or just extend the provider. Do you think there's a need for something more?

Reply:
It was more along the lines:


```dart
abstract class EvaluateHoverResult {
...
}


class DragHandlerHoverResult implements EvaluateHoverResult {
  /// If the position allows a drag and fill operation, this holds the
  /// source range for the operation.
  final Range2D? fillRange;
}
``` 

What leads me to think about this is the following doc you added:

```dart
/// Checks if the point in the table belongs to a place that should react
/// differently, like a drag and fill start position.
```

It feels like "drag and fill start position" is one of the possible results it might eventually be returned. So instead of having a single `EvaluateHoverResult`  with a bunch of nullable results, we could have this arrangement of classes.

Reply:
There are two ways to approach this: with nullable properties for the features (like the fill range, for instance), or with the different implementations of abstract classes.

Personally, I prefer the nullable in this case. Of course that in the future, if too many features are added, it will benefit from a refactor, but right now, it's a lot easier to just check if the value is null than having to go through the trouble of checking the object type and casting it.

What do you think?

Reply:
I think thats ok for now. I cant think of other features that will need this any time soon 👍🏽